modelling/src/main/java/org/axonframework/modelling/SimpleEntityAsyncRepository.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.messaging.Context.ResourceKey;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.modelling.repository.AsyncRepository;
+import org.axonframework.modelling.repository.ManagedEntity;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.UnaryOperator;
+
+/**
+ * Simple implementation of an {@link AsyncRepository} that can load and persist entities of a given type. When an
+ * entity is loaded, it is stored in the {@link ProcessingContext} to ensure that the same entity is returned when it is
+ * loaded again in the same context.
+ * <p>
+ * When an entity is successfully loaded, it's scheduled to automatically be persisted when the
+ * {@link ProcessingContext} is committed.
+ * <p>
+ * Both the {@link #load(Object, ProcessingContext)} and {@link #loadOrCreate(Object, ProcessingContext)} use the
+ * {@link SimpleEntityLoader} to load the entity. If you wish to create an entity when it does not exist, you can return
+ * a value when you have found none.
+ *
+ * @param <I> The type of the identifier of the entity.
+ * @param <T> The type of the entity.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class SimpleEntityAsyncRepository<I, T> implements AsyncRepository.LifecycleManagement<I, T> {
```

Comment:
I don't think I am completely clear on why we need this repository implementation. Would you care to explain so that I don't start filling in things for myself on this subject?

Reply:
Like discussed in the meeting, but for public documentation:
This is the most flexible type of repository there is. 
Whenever an entity is loaded through the loader, it gets attached to the ProcessingContext and the persister gets called when the UnitOfWork commits. 

It's essentially a way to let the `StateManger` only deal with repositories.